### ISSUE TITLE
The request could not be completed because: Could not set team ID to 'XXXYYY', only found the following available teams

### ISSUE THREAD

**Issue by evgeniisage** [fastlane/fastlane#16276]
### New Issue Checklist

- [X] Updated fastlane to the latest version
- [X] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [X] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [X] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
I'm trying to upload an iOS app to TestFlight using fastlane on a hosted Azure DevOps agent.
The desired workflow is as follows: 
**Step 1**
job 1) Initiate fastlane on an agent - V works as expected
**Step 2**
Job 1) Initialize jobs - V works as expected
Job 2) Install Apple Distribution certificate - V works as expected
Job 3) Download binaries from a repo - V works as expected
Job 4) Resign the app using distribution cert and provisioning profile - V works as expected
Job 5) Publish to the App Store TestFlight track - X fails with the following error message:

2020-04-07T19:52:42.9135780Z [!] The request could not be completed because:
2020-04-07T19:52:42.9136680Z 	Could not set team ID to 'XXXYYY', only found the following available teams:

2020-04-07T19:52:42.9137580Z - ZZZWWW (Company 2 - A1B23C)
2020-04-07T19:52:42.9138150Z - AABB (Company 2)
2020-04-07T19:52:42.9138750Z - CCDD (Company 3)

##### Command executed
fastlane pilot upload -u *** -i /Users/runner/runners/2.165.2/work/r1/a/xMobile/IPA/AppName.ipa -q XXXYYY -r My Company -a com.app.name

Please note, the arguments for the command are taken from Azure DevOps job inputs and not written manually.

##### Complete output when running fastlane, including the stack trace and command used

2020-04-07T19:52:19.3337350Z ##[section]Starting: Publish to the App Store TestFlight track
2020-04-07T19:52:19.3349220Z ==============================================================================
2020-04-07T19:52:19.3349700Z Task         : Apple App Store Release
2020-04-07T19:52:19.3350120Z Description  : Release an app to TestFlight or the Apple App Store
2020-04-07T19:52:19.3350460Z Version      : 1.158.0
2020-04-07T19:52:19.3350760Z Author       : Microsoft Corporation
2020-04-07T19:52:19.3351240Z Help         : [More Information](https://marketplace.visualstudio.com/items?itemName=ms-vsclient.app-store)
2020-04-07T19:52:19.3351820Z ==============================================================================
2020-04-07T19:52:19.5191520Z 244a7aa4-33a2-4b83-b92b-9f10eb254eec exists true
2020-04-07T19:52:19.5309980Z [command]/usr/local/opt/ruby/bin/gem install fastlane
2020-04-07T19:52:19.5367470Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5368430Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5369510Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5370260Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5371060Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5371800Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5372610Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5373350Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5374210Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5375050Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5375810Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5376580Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5377340Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5378130Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5378890Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5379670Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5380420Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5381220Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5381950Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5382760Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5383500Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5384290Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5385030Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5385820Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5386570Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5387380Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5388110Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5388900Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5389690Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5390440Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5391920Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:19.5392690Z (node:758) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-04-07T19:52:31.0176570Z Successfully installed fastlane-2.145.0
2020-04-07T19:52:31.0177390Z Parsing documentation for fastlane-2.145.0
2020-04-07T19:52:31.0177830Z Done installing documentation for fastlane after 7 seconds
2020-04-07T19:52:31.0178680Z 1 gem installed
2020-04-07T19:52:31.0455640Z [command]/usr/local/opt/ruby/bin/gem update fastlane -i /Users/runner/.gem-cache
2020-04-07T19:52:33.5600920Z Updating installed gems
2020-04-07T19:52:33.5601470Z Nothing to update
2020-04-07T19:52:33.5860810Z [command]fastlane pilot upload -u *** -i /Users/runner/runners/2.165.2/work/r1/a/xMobile/IPA/AppName.ipa -q XXXYYY -r My Company -a com.app.name
2020-04-07T19:52:36.2156210Z [19:52:36]: [33mGet started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile[0m
2020-04-07T19:52:39.9021010Z [19:52:39]: Login to App Store Connect (***)
2020-04-07T19:52:42.2740630Z 
2020-04-07T19:52:42.2741310Z Looking for related GitHub issues on fastlane/fastlane...
2020-04-07T19:52:42.2741640Z 
2020-04-07T19:52:42.9056020Z ➡️  Possible to pass fastlane a .app or .zip?
2020-04-07T19:52:42.9056890Z     https://github.com/fastlane/fastlane/issues/330 [closed] 2 💬
2020-04-07T19:52:42.9124640Z     22 Sep 2016
2020-04-07T19:52:42.9125010Z 
2020-04-07T19:52:42.9126630Z ➡️  s3 action failure after migrating to 1.3.0
2020-04-07T19:52:42.9127900Z     https://github.com/fastlane/fastlane/issues/318 [closed] 6 💬
2020-04-07T19:52:42.9128470Z     22 Sep 2016
2020-04-07T19:52:42.9128700Z 
2020-04-07T19:52:42.9129790Z ➡️  Trouble using fastlane / sigh on multi-user system
2020-04-07T19:52:42.9131160Z     https://github.com/fastlane/fastlane/issues/326 [closed] 13 💬
2020-04-07T19:52:42.9131590Z     08 Nov 2016
2020-04-07T19:52:42.9131740Z 
2020-04-07T19:52:42.9133460Z and 16151 more at: https://github.com/fastlane/fastlane/search?q=The%20request%20could%20not%20be%20completed%20because%3A%0A%09Could%20not%20set%20team%20ID%20to%20%27117949423%27%2C%20only%20found%20the%20following%20available%20teams%3A%0A%0A-%20119531376%20%28SAGE%20SOFTWARE%2C%20INC.%20-%20J9K9638GUE%29%0A-%20300156%20%28Sage%20SP%2C%20S.L.%29%0A-%2094016%20%28Sage%20Software%2C%20Inc.%29%0A&type=Issues&utf8=✓
2020-04-07T19:52:42.9134530Z 
2020-04-07T19:52:42.9135210Z 🔗  You can ⌘ + double-click on links to open them directly in your browser.
2020-04-07T19:52:42.9135540Z 
2020-04-07T19:52:42.9135780Z [!] The request could not be completed because:
2020-04-07T19:52:42.9136680Z 	Could not set team ID to 'XXXYYY', only found the following available teams:
2020-04-07T19:52:42.9137580Z - ZZZWWW (Company 2 - A1B23C)
2020-04-07T19:52:42.9138150Z - AABB (Company 2)
2020-04-07T19:52:42.9138750Z - CCDD (Company 3)
2020-04-07T19:52:42.9311360Z ##[error]Error: fastlane failed with return code: 1
2020-04-07T19:52:42.9326220Z ##[section]Finishing: Publish to the App Store TestFlight track

I've read in similar issues that one should use Itunes Connect team id instead of Apple Developer team id and I did that. I have an environment variable called TeamNum that stores this id and I checked it using Martín De la Fuente's advice from this stack overflow issue: https://stackoverflow.com/questions/42960644/how-to-get-itunes-connect-team-id-and-team-name
So I am 100% sure team id is correct but for some reason fastlane is not accepting it. 

### Environment

Unfortunately, I can't attach details of the environment as the fastlane agent is out of my control. Thanks!

**Comment by evgeniisage** [comment]
Update: I realized that the error was due to the Azure DevOps agent user not being part of the specified team. If I add fastlane argument with usernmae and set user to my own with password being taken from environment variable (FASTLANE_PASSWORD), the upload fails with an error message that credentials seem to be incorrect. I know they are correct and remember facing the same error message on a physical Mac OS agent until I added my credentials to a keychain. Thus, the question is: is there a way to add credentials to keychain if I use a standard AzDO task group called "Publish to the App Store TestFlight track" or is there a different way to make the upload work in my setup. Will happily provide any further information on this issue. Thanks!
